sklearn/cluster/tests/test_affinity_propagation.py

```diff
@@ -195,17 +207,18 @@ def test_affinity_propagation_non_convergence_regressiontest():
     assert_array_equal(np.array([0, 0, 0]), af.labels_)
 
 
-def test_equal_similarities_and_preferences():
+@pytest.mark.parametrize("dtype", DTYPES)
+def test_equal_similarities_and_preferences(dtype):
     # Unequal distances
-    X = np.array([[0, 0], [1, 1], [-2, -2]])
+    X = np.array([[0, 0], [1, 1], [-2, -2]]).astype(dtype)
     S = -euclidean_distances(X, squared=True)
 
```

Comment:
To make sure that the `_equal_similarities_and_preferences` is actually running on a dtype'ed input.

```suggestion
    assert S.dtype == dtype

```

Reply:
This is already tested by euclidean_distances